Website design

cb341 opened the pull request:
# Website styling

## Features

- navbar
- intro section
- technology section
- recommendations section
- comparison section placeholder
- support section
- very basic footer

<details>
<summary>Screenshots</summary>



</details>

<details>
<summary>Canva page</summary>

https://deploio.my.canva.site
![Screenshot 2023-10-17 at 16 39 01](https://github.com/renuo/deplo.io-landing-page/assets/53896675/cd513733-b113-4033-9b2c-4f409bb592ba)

</details>

cb341 committed:
Add navigation bar

cb341 committed:
Update styling

cb341 committed:
Refactor stylesheet, add static assets

cb341 committed:
Align deplo.io logo

cb341 committed:
Add tech images, design changes

cb341 committed:
Update certificate / recommendations section

cb341 committed:
Add google tags

cb341 committed:
Update certificate section

cb341 committed:
Refactor recomendation section

cb341 committed:
Add comparison section

cb341 committed:
Add support section

cb341 committed:
Add basic footers

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+
+.content-wrapper {
+    max-width: 100vw;
+    overflow-x: visible;
```
Isn't `overflow-x` visible by default?

cb341 replied:
true, for debug purposes I unset it from `hidden`...
Thanks for noticing!

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+}
+
+.seo {
+    display: none;
```
If you want an element to only be visible for screen readers you could use something like this:
```css
clip: rect(1px 1px 1px 1px)
height: 1px
overflow: hidden
width: 1px
```
otherwise it'll be ignored

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+    align-items: center;
+    z-index: 1;
+
+    &, & * {
```
nesting [still isn't supported everywhere](https://caniuse.com/css-nesting)

cb341 replied:
I know, should I default to SASS and transpile it to css?

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+
+    & .logo-container {
+        justify-content: center;
+        align-items: center;
```
you are using these `center` lines for multiple elements. you could create a new class `.center` instead

cb341 replied:
Sure, this could be refactored in Sass with a mixin. I am still thinking about migrating to scss.

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+
+.section-recommendations {
+    flex-direction: row;
+    gap: 75px;
```
don't hard-code such a large gap

cb341 replied:
What should I do otherwise?

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+}
+
+.list {
+}
```
unused code

cb341 replied:
Thx

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+    background: url("../../assets/images/Swiss Flag.png");
+    background-size: 100%;
+    flex-direction: row;
+    --margin: calc(-1 * calc(100vw - var(--content-width)) / 2);
```
the 2nd calc is redundant (i.e. `calc(-1 * (100vw - var(--content-width)) / 2)`) but this approach feels kinda hacky anyway

cb341 replied:
It is, might as well consider adding `reduced-width` class to all sections that shouldn't take up 100% of the viewport width.

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+
+.flex-row {
+    flex-direction: row;
+}
```
if you have ~drills~ utils for this you could use them for e.g. `.certificate` instead of writing the line again

cb341 replied:
Drills?

hunchr replied:
*utils for these classes (sry, autocorrect)

hunchr commented on src/css/style.css:
```diff
@@ -1,3 +1,373 @@
+    padding: 1.5rem;
+    font-size: 2em;
+    border-radius: 20px;
+}
```
is there a reason why you switch between `rem`, `em` and `px`?

cb341 replied:
Not in particular, no haha

hunchr commented on src/index.html:
```diff
@@ -7,8 +7,205 @@
+        </header>
+
+        <section class="section-intro">
+            <img class="img img-wave-logo" src="../assets/images/deploio_wave_logo.png">
```
`alt` is missing

cb341 replied:
Thanks for noticing

hunchr reviewed (changes requested):
the website isn't responsive but I assume you're still working on that

cb341 committed:
Update project structure, add sass

cb341 committed:
Refactor stylesheet with scss

cb341 committed:
Update navigation semantics

cb341 committed:
Add image labels

schmijos committed:
Create LICENSE

cb341 committed:
Attempt deplo.io rewrite

cb341 committed:
Add skeleton, basic styling

cb341 committed:
Update content blocks, improve responsiveness

cb341 committed:
Replace original site, with the responsive version

cb341 committed:
Refactor stylesheet

- refactor cards
- remove debug stylings for sections

cb341 committed:
Update colors, add buttons

cb341 committed:
Rename tech logos, add them to the dom

cb341 committed:
Add certificate section

cb341 committed:
Add 'hosted_on_deplo.io'

cb341 committed:
Refactor stylesheet

samuelsteiner committed:
Update README.md

cb341 committed:
Polish styles

cb341 committed:
Merge pull request #2 from renuo/feature/rewrite

Feature/rewrite

cb341 committed:
Fix styling issues

- Background image not covering entire element (section-support::after)
- Form elements don't inherit parent font size
- Unused css in comments should be removed
- Margin inside contact form should be replaced with a gap

cb341 committed:
Refactor markup

- Remove unused html
- Replace template string for welcome message with template element
- FIx footer-2 aligment
- Translate contact form

samuelsteiner committed:
Pretend to be PHP to trick Deploio

samuelsteiner committed:
Merge branch 'develop' of https://github.com/renuo/deploio-landing-page into develop

cb341 committed:
Add meta tags

cb341 committed:
Add meta images, manifest

cb341 committed:
Fix links, update btn size in mobile view

cb341 committed:
Update footer logos, so that they are next to each other in mobile

firemind commented on public/assets/images/language_ruby.png:
@coorasse didn't you mention something about the rails logo having to stay red? Or does this not apply here?

cb341 replied:
Thanks for the info, I'll quickly discuss this with Yessin.

firemind commented on public/index.html:
```diff
@@ -0,0 +1,383 @@
+            </div>
+        </div>
+        <div>
+            <!--            <a href="#"><img src="assets/images/swiss.png" alt="swiss made, swiss hosting"></a>-->
```
Why is this commented out. do we need it?

cb341 replied:
No, we no longer do.

firemind commented on public/index.html:
```diff
@@ -0,0 +1,383 @@
+    </div>
+</footer>
+</body>
+</html>
```
Putting this page through https://validator.w3.org/nu/#textarea shows a couple of pretty basic improvements you can make (like adding a `<title>` and `alt` attributes on images.

cb341 replied:
# SEO, screenreaders

I added the title tag, alt attributes, but I don't know what's up with the article headings.

The outline looks correct:
![CleanShot 2023-10-23 at 15 55 18@2x](https://github.com/renuo/deploio-landing-page/assets/53896675/8a612077-c418-4a11-8da5-e44b2b475e31)
![CleanShot 2023-10-23 at 15 55 10@2x](https://github.com/renuo/deploio-landing-page/assets/53896675/a43709fa-2382-452e-b44d-6da50a26cbb4)

## Question

The article is sections into `section` elements, each with its own `h2, h3` tags. Is the `h1` not allowed to be inside a `section` tag?

firemind replied:
Yes, it is recommended to have exactly one `h1` per page and have that be above the `section` elements. The sections sub-divide the document while the `h1` is supposed to be the title of the entire document.

cb341 replied:
Alright, I replaced the `<section>` tag with a `<div>` and this has improved the whole situation.

cb341 replied:
![CleanShot 2023-10-23 at 16 10 06@2x](https://github.com/renuo/deploio-landing-page/assets/53896675/c59bcb12-62d7-4b71-9628-59df793bad8a)

firemind commented on src/style.scss:
```diff
@@ -0,0 +1,472 @@
+    .footer-providers {
+      //display: flex;
+      //align-items: center;
+      //justify-content: center;
```
Commented out code

firemind commented on public/index.html:
```diff
@@ -0,0 +1,383 @@
+        >Jetzt Testen</a>
+        <div>
+            <a
+                    href="https://cockpit.nine.ch/en/session/new?origin=%2F"
```
This entire page is in german but the "Sign in" and "Sign Up" are in english and this link points to the english sign in page.

cb341 replied:
Thanks for noticing. The navigation elements will remain in english, but I will update the link.

cb341 committed:
Update design

- Reduce vertical spacing in footer when on mobile
- Add alt tags for better accessibility
- Refactor `/>` to `>` for images, links

cb341 committed:
Add margin to copyright element in footer

cb341 committed:
Add missing alt

cb341 committed:
Add comparison table

cb341 committed:
Update cockpit link, add h2 seo tag to comparsion section

cb341 committed:
Replace introduction section wrapper with a div
